### ISSUE TITLE
Can not decode ActiveSupport::TimeWithZone for certain time zones

### ISSUE THREAD

**Issue by owaiswiz** [hopsoft/universalid#34]
I see when packing, the extension uses `ActiveSupport::TimeWithZone#zone` which is the abbreviation of the zone name:
https://github.com/hopsoft/universalid/blob/c0ac50bf09425bca205b00bc6287035ba18dbfbc/lib/universalid/extensions/active_support/time_with_zone_message_pack_type.rb#L9

And during unpacking it uses this value to get the zone:
https://github.com/hopsoft/universalid/blob/c0ac50bf09425bca205b00bc6287035ba18dbfbc/lib/universalid/extensions/active_support/time_with_zone_message_pack_type.rb#L14

That works for some timezones. But does not work in all cases.

To reproduce:

```ruby
#============= Works ==============
Time.zone = 'EST'
Time.zone.inspect #=> "#<ActiveSupport::TimeZone:0x0000000285ebb450 @name=\"EST\", @utc_offset=nil, @tzinfo=#<TZInfo::DataTimezone: EST>>"
Time.current.zone => "EST"
Time.current.time_zone.name #=> "EST"
ActiveSupport::TimeZone[Time.current.time_zone.name] #=> #<ActiveSupport::TimeZone:0x0000000285ebb450 @name="EST", @tzinfo=#<TZInfo::DataTimezone: EST>, @utc_offset=nil>
ActiveSupport::TimeZone[Time.current.zone] #=> #<ActiveSupport::TimeZone:0x0000000285ebb450 @name="EST", @tzinfo=#<TZInfo::DataTimezone: EST>, @utc_offset=nil>

URI::UID.parse(URI::UID.build(Time.current).to_s).decode.zone
#=> "EST"


#========= Does not work ========== 
Time.zone = 'Indian/Chagos'
Time.zone.inspect  # => "#<ActiveSupport::TimeZone:0x0000000176335a28 @name=\"Indian/Chagos\", @utc_offset=nil, @tzinfo=#<TZInfo::DataTimezone: Indian/Chagos>>"
Time.current.zone #=> "+06"
Time.current.time_zone.name #=> "Indian/Chagos"
ActiveSupport::TimeZone[Time.current.time_zone.name] #=> #<ActiveSupport::TimeZone:0x0000000176335a28 @name="Indian/Chagos", @tzinfo=#<TZInfo::DataTimezone: Indian/Chagos>, @utc_offset=nil>
ActiveSupport::TimeZone[Time.current.zone] #=> nil

URI::UID.parse(URI::UID.build(Time.current).to_s).decode.zone
#=> NoMethodError: undefined method `period_for_utc' for nil:NilClass
#      from /Users/owaiskhan/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-
#      7.0.8/lib/active_support/time_with_zone.rb:83:in `period'
#      Caused by NoMethodError: undefined method `period_for_utc' for nil:NilClass


```

I am wondering if it's better to change `packer.write obj.zone` with `packer.write obj.time_zone.name`

**Comment by owaiswiz** [comment]
Even the example zone above that "works" won't always work BTW. 

E.g when the time is during a period of daylight savings `time.zone` would return `EDT`.

Which again would not work when you try to unpack it because `ActiveSupport::TimeZone['EDT']` is nil.

**Comment by owaiswiz** [comment]
Also noticed another thing. The first argument to `ActiveSupport::TimeWithZone.new time, ActiveSupport::TimeZone[zone]` must be in UTC from what I am seeing.


Incorrect (what we are doing now?):
```ruby
Time.current;
#=> 2024-01-25 09:53:53 -0500
ActiveSupport::TimeWithZone.new Time.parse(Time.current.iso8601(9)), ActiveSupport::TimeZone[Time.current.zone]
#=> 2024-01-25 04:53:53 -0500
```

Correct:
```ruby
puts Time.current;
#=> 2024-01-25 09:53:47 -0500
puts ActiveSupport::TimeWithZone.new Time.parse(Time.current.iso8601(9)).utc, ActiveSupport::TimeZone[Time.current.zone]
#=> 2024-01-25 09:53:47 -0500
```

**Comment by hopsoft** [comment]
Thanks for this report. I'll get this fixed up.

**Comment by hopsoft** [comment]
Fix shipped with v0.1.6. Let me know if you find any issues going forward.

**Comment by owaiswiz** [comment]
Thank you! Will do. Loving this library.
